content/en/cosign/openid_signing.md

```diff
@@ -35,6 +35,26 @@ To log in and set up your OIDC Identity, follow these steps:
 
 Currently, Sigstore supports Microsoft, Google, and GitHub. As an alternative, you can also use the environment variable `SIGSTORE_ID_TOKEN` to identify yourself by setting its value to that of the identity token. Cosign also has support for detecting some of these automated environments and producing an identity token. Currently this supports Google and GitHub, in addition to other environments. See [Cosign's providers](https://github.com/sigstore/cosign/tree/main/pkg/providers) for a complete list.
 
+### The signing, witnessing, and verifying process
+
+When taking advantage of sigstore’s full capabilities, the process of signing is as follows:
+
+#### Verifying identity and signing the artifact
+
+1) A keypair is created. 
+2) Sigstore checks the identity of the user signing the artifact and issues a certificate attesting to their identity. 
+3) The certificate is bound to the public key, which adds an extra layer of verification. Exchanging the keys will prove the identity of the private keyholder. 
+4) As the artifact is signed, the keys are exchanged, confirming the signer’s identity. 
+5) For security, the private key is destroyed shortly after and the short-lived identity certificate expires. Users who wish to verify the software will use the transparency log entry, rather than relying on the signer to safely store and manage the private key.
+
+#### Witnessing and recording the process
+
+To create the transparency log entry, Sigstore creates an object containing information that will allow signature verification without the (destroyed) private key. The object contains the hash of the artifact, the public key, and the signature. Crucially, this object is timestamped. The Rekor transparency log "witnesses" the signing event by entering a timestamped entry into the records that attests that the secure signing process has occurred. The software creator publishes the timestamped object, including the hash of the artifact, public key, and signature.
```

Comment:
Can we also add a line about "Clients upload signing events to the transparency log so that the events are publicly auditable. Artifact owners should monitor the log for their identity to verify each occurrence."